src/pages-and-resources/discussions/app-list/AppList.test.jsx

```diff
@@ -1,21 +1,124 @@
+import initializeStore from '../../../store';
+import { fetchApps } from '../data/thunks';
+import { executeThunk } from '../data/redux.test';
+import { getAppsUrl } from '../data/api';
+import { legacyApiResponse, piazzaApiResponse } from '../factories/mockApiResponses';
+import { updateStatus, LOADED, selectApp } from '../data/slice';
+
+const courseId = 'course-v1:edX+TestX+Test_Course';
+
 describe('AppList', () => {
-  test('displays a message when there are no apps available', () => {
+  let axiosMock;
+  let store;
+  let container;
+
+  function createComponent(screenWidth = 1280) {
+    return (
+      <AppProvider store={store}>
+        <ResponsiveContext.Provider value={{ width: screenWidth }}>
+          <IntlProvider locale="en" messages={{}}>
+            <AppList />
+          </IntlProvider>
+        </ResponsiveContext.Provider>
+      </AppProvider>
+    );
+  }
 
+  beforeEach(async () => {
+    initializeMockApp({
+      authenticatedUser: {
+        userId: 3,
+        username: 'abc123',
+        administrator: true,
+        roles: [],
+      },
+    });
+
+    store = await initializeStore();
+    axiosMock = new MockAdapter(getAuthenticatedHttpClient());
   });
 
-  test('display a card for each available app', () => {
+  afterEach(() => {
 
   });
 
-  test('displays the FeaturesTable at desktop sizes', () => {
+  test('displays a message when there are no apps available', async () => {
+    // selects an app to mock store data
+    store.dispatch(selectApp({ appId: 'legacy' }));
+    // update store status to LOADED
+    store.dispatch(updateStatus({ status: LOADED }));
+    // renders component
+    const component = createComponent();
+    const wrapper = render(component);
+    container = wrapper.container;
+    // assert test case results
+    expect(queryByText(container, 'There are no discussions providers available for your course.')).toBeInTheDocument();
+  });
 
+  test('displays loading state when there is no selected App', () => {
+    // render component without mocking data in store
+    const component = createComponent();
+    const wrapper = render(component);
+    container = wrapper.container;
+    expect(queryByText(container, 'Loading...')).toBeInTheDocument();
   });
 
-  test('hides the FeaturesTable at mobile sizes', () => {
+  test('display a card for each available app', async () => {
+    // mock data in redux store
+    axiosMock.onGet(getAppsUrl(courseId)).reply(200, piazzaApiResponse);
+    await executeThunk(fetchApps(courseId), store.dispatch);
+    // render componenet
+    const component = createComponent();
+    const wrapper = render(component);
+    container = wrapper.container;
+    // assert test results
+    const appCount = store.getState().discussions.appIds.length;
+    const nodeCount = queryAllByRole(container, 'radio').length;
+    expect(nodeCount === appCount);
+  });
 
+  test('displays the FeaturesTable at desktop sizes', async () => {
```

Comment:
we should also have a test for `FeaturesTable` that it is hidden on small screens.